keyvalue/clients/src/main/java/tachyon/client/keyvalue/BaseKeyValueStoreReader.java

```diff
@@ -101,4 +101,13 @@ public ByteBuffer get(ByteBuffer key) throws IOException, TachyonException {
   public KeyValueIterator iterator() throws IOException, TachyonException {
     return new KeyValueStoreIterator(mPartitions);
   }
+
+  @Override
+  public int size() throws IOException, TachyonException {
+    int totalSize = 0;
+    for (PartitionInfo partition : mPartitions) {
+      totalSize += KeyValuePartitionReader.Factory.create(partition.getBlockId()).size();
```

Comment:
close partition reader?

also as a TODO, we could move size of each partition into `PartitionInfo`